src/viadot/sources/sftp.py

```diff
@@ -0,0 +1,296 @@
+
+    def __init__(
+        self,
+        *args,
+        credentials: SftpCredentials | None = None,
+        config_key: str = "sftp",
+        **kwargs,
+    ):
+        """Create an instance of SFTP.
+
+        Args:
+            credentials (SftpCredentials, optional): SFTP credentials. Defaults to None.
+            config_key (str, optional): The key in the viadot config holding relevant
+                credentials. Defaults to "sftp".
+
+        Notes:
+            self.conn is paramiko.SFTPClient.from_transport method that contains
+            additional methods like get, put, open etc. Some of them were not
+            implemented in that class. For more check documentation
+            (https://docs.paramiko.org/en/stable/api/sftp.html).
+
+            sftp = SftpConnector()
+            sftp.conn.open(filename='folder_a/my_file.zip', mode='r')
+
+        Raises:
+            CredentialError: If credentials are not provided in local_config or
+                directly as a parameter.
+        """
+        credentials = credentials or get_source_credentials(config_key) or None
+
+        if credentials is None:
+            message = "Missing credentials."
+            raise CredentialError(message)
+        self.credentials = credentials
+
+        validated_creds = dict(SftpCredentials(**credentials))
+        super().__init__(*args, credentials=validated_creds, **kwargs)
+
+        self.conn = None
+        self.hostname = validated_creds.get("hostname")
+        self.username = validated_creds.get("username")
+        self.password = validated_creds.get("password")
+        self.port = validated_creds.get("port")
+        self.rsa_key = validated_creds.get("rsa_key")
+
+    def _getfo_file(self, file_name: str) -> BytesIO:
+        """Copy a remote file from the SFTP server and write to a file-like object.
+
+        Args:
+            file_name (str, optional): File name to copy.
+
+        Returns:
+            BytesIO: file-like object.
+        """
+        flo = BytesIO()
+        try:
+            self.conn.getfo(file_name, flo)
+
+        except FileNotFoundError as error:
+            raise SFTPError from error
+
+        else:
+            return flo
+
+    def get_connection(self) -> paramiko.SFTPClient:
+        """Returns a SFTP connection object.
+
+        Returns: paramiko.SFTPClient.
+        """
+        ssh = paramiko.SSHClient()
+
+        if len(self.rsa_key) == 0:
+            transport = paramiko.Transport((self.hostname, self.port))
+            transport.connect(None, self.username, self.password)
+
+            self.conn = paramiko.SFTPClient.from_transport(transport)
+
+        else:
+            keyfile = StringIO(self.rsa_key)
+            mykey = paramiko.RSAKey.from_private_key(keyfile)
+            ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            ssh.connect(self.hostname, username=self.username, pkey=mykey)
+            time.sleep(1)
```

Comment:
Can you add a comment in the code to explain why you're waiting here?

Reply:
No Idea, ask the connector creator.  Another backlog task.

Reply:
https://github.com/dyvenia/viadot/issues/1063